### DISCUSSION TITLE
Calling interface functions from truffle console

### DISCUSSION THREAD

**Discussion by pclaesen** [ConsenSys-archive/truffle#5514] [General Q&A]
I forked mainnet, using truffle console now to test.
For my contract, I need to approve an ERC20 token, before I can use `transferFrom()`, and I'd also like to use the `balanceOf(address)` and `allowance` function

Here is some code:
`interface IErc20 {
    function approve(address, uint256) external returns (bool);
    function transfer(address, uint256) external returns (bool);
    function transferFrom(address _from, address _to, uint _value) external returns (bool success);
    function allowance(address owner, address spender) external view returns (uint256);
    function balanceOf(address) external returns (uint256);
}`

Does anyone know how I can call these interface functions?

**Comment by apdan510** [comment]
Hey @pclaesen please take a look at the Truffle console commands here: https://trufflesuite.com/docs/truffle/reference/truffle-commands/

You will be able to interact and call functions from your contract when you enter Truffle Develop mode. Let me know if this helps.

Kind regards,
Alex | Truffle Support
